File: app/src/main/java/com/example/photoeditor/Adapter/ThumbnailAdapter.java
package com.example.photoeditor.Adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.example.photoeditor.FiltersListFragment;
import com.example.photoeditor.Interface.FilterListFragmentListener;
import com.example.photoeditor.R;
import com.zomato.photofilters.utils.ThumbnailItem;

import java.util.List;

public class ThumbnailAdapter extends RecyclerView.Adapter<ThumbnailAdapter.MyViewBolder> {

    private List<ThumbnailItem> thumbnailItems;
    private FilterListFragmentListener listener;
    private Context context;
    private int selectedIndex = 0;

    public ThumbnailAdapter(List<ThumbnailItem> thumbnailItems, FilterListFragmentListener listener, Context context) {
        this.thumbnailItems = thumbnailItems;
        this.listener = listener;
        this.context = context;
    }

    @NonNull
    @Override
    public MyViewBolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(context).inflate(R.layout.thumbnail_item, parent, false);
        return new MyViewBolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewBolder holder, final int position) {
        final ThumbnailItem thumbnailItem = thumbnailItems.get(position);

        holder.thumbnail.setImageBitmap(thumbnailItem.image);
        holder.thumbnail.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                listener.onFilterSelected(thumbnailItem.filter);
                selectedIndex = position;
                notifyDataSetChanged();
            }
        });

        holder.filter_name.setText(thumbnailItem.filterName);

        if (selectedIndex == position)
            holder.filter_name.setTextColor(ContextCompat.getColor(context, R.color.selected_filter));
        else
            holder.filter_name.setTextColor(ContextCompat.getColor(context, R.color.normal_filter));
    }

    @Override
    public int getItemCount() {
        return thumbnailItems.size();
    }

    public class MyViewBolder extends RecyclerView.ViewHolder {
        ImageView thumbnail;
        TextView filter_name;
        public MyViewBolder(View itemView) {
            super(itemView);
            thumbnail = (ImageView)itemView.findViewById(R.id.thumbnail);
            filter_name = (TextView)itemView.findViewById(R.id.filter_name);
        }
    }
}

File: app/src/main/java/com/example/photoeditor/Adapter/ThumbnailAdapter.kt
package com.example.photoeditor.Adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.example.photoeditor.Adapter.ThumbnailAdapter.MyViewBolder
import com.example.photoeditor.FiltersListFragment
import com.example.photoeditor.R
import com.zomato.photofilters.utils.ThumbnailItem

class ThumbnailAdapter(private val thumbnailItems: MutableList<ThumbnailItem>, private val listener: FiltersListFragment, private val context: Context) : RecyclerView.Adapter<MyViewBolder?>() {
    private var selectedIndex = 0
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewBolder {
        val itemView = LayoutInflater.from(context).inflate(R.layout.thumbnail_item, parent, false)
        return MyViewBolder(itemView)
    }

    override fun onBindViewHolder(holder: MyViewBolder, position: Int) {
        val thumbnailItem = thumbnailItems[position]
        holder.thumbnail?.setImageBitmap(thumbnailItem.image)
        holder.thumbnail?.setOnClickListener {
            listener.onFilterSelected(thumbnailItem.filter)
            selectedIndex = position
            notifyDataSetChanged()
        }
        holder.filter_name?.text = thumbnailItem.filterName
        if (selectedIndex == position) holder.filter_name?.setTextColor(ContextCompat.getColor(context, R.color.selected_filter)) else holder.filter_name?.setTextColor(ContextCompat.getColor(context, R.color.normal_filter))
    }

    override fun getItemCount(): Int {
        return thumbnailItems.size
    }

    inner class MyViewBolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var thumbnail: ImageView? = itemView.findViewById<View?>(R.id.thumbnail) as ImageView?
        var filter_name: TextView? = itemView.findViewById<View?>(R.id.filter_name) as TextView?

    }

}